### PR TITLE
using consumer name as consumer external id

### PR DESCRIPTION
**Pull request by skeeey**
- Treat the consumer name as a consumer external id, the consumer name
   -  must be unique and not null
   - using RFC 1123 as its format 
   - cannot be updated
   - when creating a consumer, if the name is not specified, the consumer id will be used as name.
- Change the resource fk from  consumer id to consumer name

### CODE REVIEW

**Comment by skeeey** [comment]
maestro git:(eid-2) make test
OCM_ENV=testing gotestsum --jsonfile-timing-events=/Users/liuwei/go/src/github.com/openshift-online/maestro/unit-test-results.json --format short-verbose -- -p 1 -v  \
		./pkg/... \
		./cmd/...
PASS pkg/api.TestEncodeManifest/empty (0.00s)
PASS pkg/api.TestEncodeManifest/valid (0.00s)
PASS pkg/api.TestEncodeManifest (0.00s)
PASS pkg/api.TestDecodeManifest/empty (0.00s)
PASS pkg/api.TestDecodeManifest/valid (0.00s)
PASS pkg/api.TestDecodeManifest (0.00s)
PASS pkg/api.TestDecodeManifestBundle/empty (0.00s)
PASS pkg/api.TestDecodeManifestBundle/valid (0.00s)
PASS pkg/api.TestDecodeManifestBundle (0.00s)
PASS pkg/api.TestDecodeStatus/empty (0.00s)
PASS pkg/api.TestDecodeStatus/valid (0.00s)
PASS pkg/api.TestDecodeStatus (0.00s)
PASS pkg/api
EMPTY pkg/api/openapi
EMPTY pkg/api/presenters
EMPTY pkg/auth
EMPTY pkg/client/cloudevents
EMPTY pkg/client/ocm
PASS pkg/config.TestConfigReadStringFile (0.00s)
PASS pkg/config.TestConfigReadIntFile (0.00s)
PASS pkg/config.TestConfigReadBoolFile (0.00s)
PASS pkg/config.TestConfigReadQuotedFile (0.00s)
PASS pkg/config
PASS pkg/controllers.TestControllerFramework (0.00s)
PASS pkg/controllers
EMPTY pkg/dao
EMPTY pkg/dao/mocks
EMPTY pkg/db
EMPTY pkg/db/db_context
EMPTY pkg/db/db_session
EMPTY pkg/db/migrations
EMPTY pkg/db/mocks
EMPTY pkg/db/transaction
EMPTY pkg/dispatcher
PASS pkg/errors.TestErrorFormatting (0.00s)
PASS pkg/errors.TestErrorFind (0.00s)
PASS pkg/errors
EMPTY pkg/event
EMPTY pkg/handlers
EMPTY pkg/logger
PASS pkg/services.TestSQLTranslation (0.02s)
PASS pkg/services.TestResourceFindByConsumerID (0.00s)
PASS pkg/services.TestCreateInvalidResource (0.00s)
PASS pkg/services.TestValidateConsumer/validated (0.00s)
PASS pkg/services.TestValidateConsumer/wrong_name (0.00s)
PASS pkg/services.TestValidateConsumer/max_length (0.00s)
PASS pkg/services.TestValidateConsumer (0.00s)
PASS pkg/services.TestValidateNewManifest/validated_single_manifest (0.00s)
PASS pkg/services.TestValidateNewManifest/validated_bundle_manifest (0.00s)
PASS pkg/services.TestValidateNewManifest/invalidated_single_manifest (0.00s)
PASS pkg/services.TestValidateNewManifest/invalidated_bundle_manifest (0.00s)
PASS pkg/services.TestValidateNewManifest/invalidated_resource_type (0.00s)
PASS pkg/services.TestValidateNewManifest (0.00s)
PASS pkg/services.TestValidateNewObject/validated (0.00s)
PASS pkg/services.TestValidateNewObject/no_apiVersion (0.00s)
PASS pkg/services.TestValidateNewObject/no_kind (0.00s)
PASS pkg/services.TestValidateNewObject/no_name (0.00s)
PASS pkg/services.TestValidateNewObject/has_generate_name (0.00s)
PASS pkg/services.TestValidateNewObject/has_resource_version (0.00s)
PASS pkg/services.TestValidateNewObject/has_deletion_grace_period_seconds (0.00s)
PASS pkg/services.TestValidateNewObject/wrong_namespace (0.00s)
PASS pkg/services.TestValidateNewObject/wrong_api_version_(no_version) (0.00s)
PASS pkg/services.TestValidateNewObject/wrong_api_version_(no_version)#01 (0.00s)
PASS pkg/services.TestValidateNewObject/wrong_api_version (0.00s)
PASS pkg/services.TestValidateNewObject (0.00s)
PASS pkg/services.TestValidateUpdateManifest/validated_single_manifest (0.00s)
PASS pkg/services.TestValidateUpdateManifest/validated_bundle_manifest (0.00s)
PASS pkg/services.TestValidateUpdateManifest/invalidated_single_manifest (0.00s)
PASS pkg/services.TestValidateUpdateManifest/invalidated_bundle_manifest (0.00s)
PASS pkg/services.TestValidateUpdateManifest/invalidated_resource_type (0.00s)
PASS pkg/services.TestValidateUpdateManifest (0.00s)
PASS pkg/services.TestValidateUpdateObject/validated (0.00s)
PASS pkg/services.TestValidateUpdateObject/apiVersion_mismatch (0.00s)
PASS pkg/services.TestValidateUpdateObject/kind_mismatch (0.00s)
PASS pkg/services.TestValidateUpdateObject/name_mismatch (0.00s)
PASS pkg/services.TestValidateUpdateObject/namespace_mismatch (0.00s)
PASS pkg/services.TestValidateUpdateObject (0.00s)
PASS pkg/services
EMPTY pkg/util
EMPTY cmd/maestro
EMPTY cmd/maestro/agent
PASS cmd/maestro/environments.TestLoadServices (0.32s)
PASS cmd/maestro/environments
EMPTY cmd/maestro/migrate
EMPTY cmd/maestro/servecmd
EMPTY cmd/maestro/server
EMPTY cmd/maestro/server/logging

DONE 57 tests in 15.157s

make test-integration
OCM_ENV=testing gotestsum --jsonfile-timing-events=/Users/liuwei/go/src/github.com/openshift-online/maestro/integration-test-results.json --format short-verbose -- -p 1 -ldflags -s -v -timeout 1h  \
			./test/integration
I0321 10:11:02.177825   75094 framework.go:76] Initializing testing environment
I0321 10:11:02.403448   75094 framework.go:161] Using Mock OCM Authz Client
I0321 10:11:02.406832   75094 framework.go:199] Disabling Sentry error reporting
I0321 10:11:02.434147   75094 grpc_server.go:70] Serving gRPC service without TLS at 8090
I0321 10:11:02.434460   75094 healthcheck_server.go:56] Serving HealthCheck without TLS at 8083
PASS test/integration.TestConsumerGet (0.34s)
PASS test/integration.TestConsumerPost (0.31s)
PASS test/integration.TestConsumerPatch (0.47s)
PASS test/integration.TestConsumerPaging (0.33s)
PASS test/integration.TestControllerRacing (2.12s)
PASS test/integration.TestControllerReconcile (1.38s)
PASS test/integration.TestControllerSync (1.35s)
PASS test/integration.TestPulseServer (4.31s)
PASS test/integration.TestResourceGet (0.32s)
PASS test/integration.TestResourcePost (2.41s)
PASS test/integration.TestResourcePatch (0.46s)
PASS test/integration.TestResourcePaging (0.50s)
PASS test/integration.TestResourceListSearch (0.46s)
PASS test/integration.TestUpdateResourceWithRacingRequests (2.59s)
PASS test/integration.TestResourceFromGRPC (3.40s)
E0321 10:11:23.222084   75094 logger.go:129]   Unable to get all maestro instances: pq: relation "server_instances" does not exist
I0321 10:11:23.233902   75094 api_server.go:157] Web server terminated
PASS test/integration

DONE 15 tests in 26.185s

make e2e-test
./test/e2e/setup/e2e_teardown.sh
+ command -v kind
+ kind delete cluster --name maestro
Deleting cluster "maestro" ...

Random Seed: 1710984308

Will run 5 of 5 specs
•••••

Ran 5 of 5 Specs in 4.156 seconds
SUCCESS! -- 5 Passed | 0 Failed | 0 Pending | 0 Skipped
PASS

Ginkgo ran 1 suite in 10.342980682s
Test Suite Passed

**Review by morvencao — COMMENTED** [review]
/lgtm

**Review by clyang82 — APPROVED** [review]
LGTM

File: pkg/db/migrations/202311151850_add_resources.go [hunk 1]
```diff
@@ -10,9 +10,9 @@ import (
 func addResources() *gormigrate.Migration {
 	type Resource struct {
 		Model
-		Source     string `gorm:"index"`
-		ConsumerID string `gorm:"index"`
-		Version    int    `gorm:"not null"`
+		Source       string `gorm:"index"`
+		ConsumerName string `gorm:"index"`
+		Version      int    `gorm:"not null"`
```
[the remark this answers is not in the document]
Comment (skeeey) [reply pkg/db/migrations/202311151850_add_resources.go:15]: yes, we don't have a live instance now, when we have one, we must consider the migration compatibility

**Pull request merged**